### ISSUE TITLE
Libreswan: Ensure IPSEC tunnel is not terminated when the Gateway POD is going down

### ISSUE THREAD

**Issue by mangelajo** [submariner-io/submariner#386]

**Comment by nyechiel** [comment]
@sridhargaddam is this specific to strongswan kernel module? can we ensure we also cover libreswan?

**Comment by sridhargaddam** [comment]
> @sridhargaddam is this specific to strongswan kernel module? can we ensure we also cover libreswan?

AFAIU its not related to kernel module. It might be related to the way we are starting strongswan (i.e., Charon) and that might need some changes (or we may have to instruct K8s not to kill the Charon process when the corresponding POD is killed). Will be able to provide more details when I look into this issue closely.

**Comment by mangelajo** [comment]
We should test this with libreswan now :-)

**Comment by sridhargaddam** [comment]
> We should test this with libreswan now :-)

Verified the behavior with Libreswan and its same as Strongswan (i.e., when submariner-engine pod is brought down, it removes the tunnels). The tunnels get reestablished once the submariner-engine comes up.

**Comment by sridhargaddam** [comment]
I have few other things on my plate and I'm not sure if I can dedicate time to look into this use-case, so I'm assigning myself. If something has bandwidth and interested to look into it, please go ahead. Thanks.

**Comment by sridhargaddam** [comment]
Verified the behavior with Strongswan and Libreswan Cabledrivers. Unfortunately, both behave differently. 

With **Libreswan**, we see that even after the EnginePod goes down (as part of restart), the tunnels are preserved on the node so datapath is not affected while the EnginePod is down until the current IKE SAs timeout (default of 1 hour). However, when the EnginePod is coming up, there is a brief disruption to the datapath (roughly 1 to 2 secs) when Libreswan is trying to re-establish new tunnels and negotiate new SAs to the remote peers. The effect of this is that TCP connections will retransmit whereas UDP will have a small datapath loss.  

With **Strongswan**, as soon as enginePod is down, datapath is effected and is restored only after the EnginePod comes up.

There is a plan to change the default CableEngine to Libreswan once there is feature parity with strongswan. Considering that datapath is not affected with Libreswan (which is a plus point), I'm not sure if it's worth spending time debugging Strongswan datapath issue (as part of this issue) - so I modified the issue description to point to Libreswan.
